### PR TITLE
docs: clarify wlr_seat's selection signals

### PR DESCRIPTION
**Pull request by tudurom**

### CODE REVIEW

File: include/wlr/types/wlr_seat.h [hunk 1]
```diff
@@ -252,10 +252,25 @@ struct wlr_seat {
 		struct wl_signal request_set_cursor;
 
 		// wlr_seat_request_set_selection_event
```
Comment (emersion) [inline include/wlr/types/wlr_seat.h:254]: Can we move this comment to the right of `request_set_selection` and `set_selection`? This is a convention meaning the `data` listener argument is of this type.

File: include/wlr/types/wlr_seat.h [hunk 2]
```diff
@@ -252,10 +252,25 @@ struct wlr_seat {
...
+		// Called when an application _wants_ to set the selection (user copies some data).
+		// More specifically, the application wants to set a wlr_data_source object for the selection.
+		// This is what a compositor usually listens for.
 		struct wl_signal request_set_selection;
+		// Called after the requested data source is set for the selection
+		// i.e. after you call wlr_seat_set_selection.
+		// When this is called, the data is already copied. It's more of a confirmation.
 		struct wl_signal set_selection;
+
 		// wlr_seat_request_set_primary_selection_event
```
Comment (emersion) [inline include/wlr/types/wlr_seat.h:265]: Ditto

File: include/wlr/types/wlr_seat.h [hunk 3]
```diff
@@ -252,10 +252,25 @@ struct wlr_seat {
 		struct wl_signal request_set_cursor;
 
 		// wlr_seat_request_set_selection_event
+
+		// Called when an application _wants_ to set the selection (user copies some data).
+		// More specifically, the application wants to set a wlr_data_source object for the selection.
+		// This is what a compositor usually listens for.
```
Comment (emersion) [inline include/wlr/types/wlr_seat.h:258]: Maybe this is slightly better?

"Compositors should listen to this event and call wlr_seat_set_selection if they want to accept the client's request."

File: include/wlr/types/wlr_seat.h [hunk 4]
```diff
@@ -252,10 +252,25 @@ struct wlr_seat {
...
 		struct wl_signal request_set_selection;
+		// Called after the requested data source is set for the selection
+		// i.e. after you call wlr_seat_set_selection.
+		// When this is called, the data is already copied. It's more of a confirmation.
```
Comment (emersion) [inline include/wlr/types/wlr_seat.h:262]: "the data is already copied" → no, data hasn't been copied when selection is set.

"It's more of a confirmation." → this is a little bit vague

File: include/wlr/types/wlr_seat.h (hunk 1 above)
Comment (tudurom) [reply include/wlr/types/wlr_seat.h:254]: The comments are rather long, should i write them like this?

```c
	struct wl_signal request_set_selection; // lorem ipsum dolor sit amet
	                                        // this comment is indented with one tab and aligned
	                                        // with many spaces.
```

Comment (emersion) [reply include/wlr/types/wlr_seat.h:254]: I mean just the `// wlr_seat_request_set_selection_event` comment.

Comment (emersion) [reply include/wlr/types/wlr_seat.h:254]: Or just add a plain-text sentence: "The signal data is a pointer to a struct wlr_seat_request_set_selection_event."

**Comment by tudurom** [comment]
Should be more clear now

File: include/wlr/types/wlr_seat.h [hunk 5]
```diff
@@ -251,11 +251,18 @@ struct wlr_seat {
...
+		// if they want to accept the client's request.
+		struct wl_signal request_set_selection; // wlr_seat_request_set_selection_event
+		// Called after the requested data source is set for the selection.
 		struct wl_signal set_selection;
-		// wlr_seat_request_set_primary_selection_event
-		struct wl_signal request_set_primary_selection;
+
+		// Called when an application _wants_ to set the primary selection (user selects some data).
+		// Compositors should listen to this event and call wlr_seat_set_primary_selection
+		// if they wan to accept the client's request.
```
Comment (emersion) [inline include/wlr/types/wlr_seat.h:263]: Typo: want

File: include/wlr/types/wlr_seat.h [hunk 6]
```diff
@@ -251,11 +251,18 @@ struct wlr_seat {
 		// wlr_seat_pointer_request_set_cursor_event
 		struct wl_signal request_set_cursor;
 
-		// wlr_seat_request_set_selection_event
-		struct wl_signal request_set_selection;
+		// Called when an application _wants_ to set the selection (user copies some data).
+		// Compositors should listen to this event and call wlr_seat_set_selection
+		// if they want to accept the client's request.
+		struct wl_signal request_set_selection; // wlr_seat_request_set_selection_event
+		// Called after the requested data source is set for the selection.
```
Comment (emersion) [inline include/wlr/types/wlr_seat.h:258]: Let's leave "requested" out. The compositor can set the selection to a `wlr_data_source` that hasn't been requested by any client.

**Pull request merged**